client/src/graph.c

```diff
@@ -109,6 +109,20 @@ void save_restoreGB(uint8_t saveOpt) {
     }
 }
 
+// Returns a copied value of what's in the Buffer at the index supplied
+// Gets from the Operation Buffer if useGraphBuffer is false
+int32_t get_graph_value_at(size_t index, bool useGraphBuffer) {
+    int32_t copiedValue;
+
+    if(useGraphBuffer) {
+        copiedValue = g_GraphBuffer[index];
+    } else {
+        copiedValue = g_OperationBuffer[index];
+    }
+
+    return copiedValue;
+}
+
```

Comment:
you are squeezing two arrays into one methodology,      learn to understand the code so it uses  the operations buffer and then applies all changes to right buffer in use.